prodtests/full-system-test/workflow-setup.sh

```diff
@@ -55,12 +55,15 @@ TRACK_SOURCES=
 has_detectors_reco ITS TPC && has_detector_matching ITSTPC && add_comma_separated TRACK_SOURCES "ITS-TPC"
 has_detectors_reco TPC TRD && has_detector_matching TPCTRD && { add_comma_separated TRD_SOURCES TPC; add_comma_separated TRACK_SOURCES "TPC-TRD"; }
 has_detectors_reco ITS TPC TRD && has_detector_matching ITSTPCTRD && { add_comma_separated TRD_SOURCES ITS-TPC; add_comma_separated TRACK_SOURCES "ITS-TPC-TRD"; }
-has_detectors_reco TPC TOF && has_detector_matching TPCTOF && { add_comma_separated TOF_SOURCES TPC; add_comma_separated TRACK_SOURCES "TPC-TOF"; }
+has_detectors_reco TPC TOF && { add_comma_separated TOF_SOURCES TPC; add_comma_separated TRACK_SOURCES "TPC-TOF"; }
 has_detectors_reco ITS TPC TOF && has_detector_matching ITSTPCTOF && { add_comma_separated TOF_SOURCES ITS-TPC; add_comma_separated TRACK_SOURCES "ITS-TPC-TOF"; }
-has_detectors_reco TPC TRD TOF && has_detector_matching TPCTRDTOF && { add_comma_separated TOF_SOURCES TPC-TRD; add_comma_separated TRACK_SOURCES "TPC-TRD-TOF"; }
+has_detectors_reco TPC TRD TOF && has_detector_matching TPCTRD && { add_comma_separated TOF_SOURCES TPC-TRD; add_comma_separated TRACK_SOURCES "TPC-TRD-TOF"; }
```

Comment:
Hi @noferini 
in this way you have unconditionally TPC-TOF matching (provided detectors are there) while the TPC-TRD-TOF is conditioned by the TPC-TRD matching and ITS-TPC-TRD-TOF by ITS-TPC-TRD.
What is the reason of such an asymmetric request? 
Note that the TRD matching in the syncmode is limited to those IRframes which where selected by the ITS so we can downscale the amount of tracks processed by the TRD. If TOF matching it conditioned by the ITS-TPC or ITS-TPC-TRD, then the TOF is also automatically downscaled. But enabling unconditional TOF matching to TPC w/o any downscaling may be to CPU consuming at high load. Why do you need to enable it by default?

Reply:
Hi Ruben,
I tried to follow the logic I found.

For TPC I needed to assume it is always there since I cannot add "TPC" to WORKFLOW_DETECTORS_MATCHING (it triggers strange behaviors in other workflow)

For other combination I add TOF only for the combination without TOF (I added a fix for that) is included in WORKFLOW_DETECTORS_MATCHING

Reply:
But what was wrong with looking on the `has_detector_matching TPCTOF` etc ?

Reply:
Ok, I restored the check with TPCTOF.

My doubt is the following.
With this logic (checking *track*TOF instead of *track*) it is in principle possible that somebody add
TPCTRDTOF to WORKFLOW_DETECTORS_MATCHING but not TPCTRD

This will trigger an error since it will add a TOF_SOURCE TPCTRD in TOF workflow which is not provided as input since dependencies are not requested by WORKFLOW_DETECTORS_MATCHING .

Reply:
Well, you can set as a condition of TPCTRDTOF `  has_detector_matching TPCTRD && has_detector_matching TPCTRDTOF`, though there are many other ways to screw up the workflow.

But why do you want to add unconstrained TPC matching to TOF in the sync. workflow. Did you check up to which occupancy is it feasible? Note that even TPC-TRD is not enables by default, although it is able to work with prescaled data.

Reply:
I started from the same TOF matching QC json.
If we decid we are not interested to it in sync phase I can easily remove it

Reply:
Ok, I'll comment it for the moment. Is it ok?

Reply:
It is a bit more complicated. I need TPC in TOF_SOURCE for QC. I have to work a bit to decouple QC from matcher

Reply:
ok

Reply:
I just keep TPC in track sources since it is needed by QC